posydon/popsyn/normalized_pop_mass.py

```diff
@@ -107,30 +108,74 @@ def fraction_simulated(f_bin, m_1, m_2, m_min, m_max,
     m_a = kwargs['primary_mass_min']
     m_b = kwargs['primary_mass_max']
 
     f0 = 1/(quad(imf_part_1, m_min, m_1, args=(m_min, alpha1))[0]
-            + quad(imf_part_2, m_1, m_2, args=(m_1, m_min, alpha1, alpha2))[0]
-            + quad(imf_part_3, m_2, m_max, args=(m_1, m_2, m_min, alpha1,
-                                                 alpha2, alpha3))[0])
-
-    f_corr = (fraction_simulated(f_bin, m_1, m_2, m_min,
-                                 m_max, alpha1, alpha2, alpha3)
-              * mean_mass_simulated(alpha3, m_a, m_b)
-              / mean_mass_of_binary(f0, f_bin, m_1, m_2, m_min, m_max,
-                                    alpha1, alpha2, alpha3))
-
-    f_model = fraction_simulated(f_bin, m_1, m_2, m_min, m_max,
-                                 alpha1, alpha2, alpha3)
-
-    return initial_ZAMS_TOTAL_mass / f_corr, f_corr, f_model
+                + quad(imf_part_2, m_1, m_2, args=(m_1, m_min, alpha1, alpha2))[0]
+                + quad(imf_part_3, m_2, m_max, args=(m_1, m_2, m_min, alpha1, alpha2, alpha3))[0])
+
+    
+    if (f_bin_simulated == 1) & (f_bin_nature != 0): #when you have modeled only binary stars
+       
+
+        f_corr_binaries = (fraction_simulated(f_bin_nature, m_1, m_2, m_min, m_max, alpha1, alpha2, alpha3)
+                  * mean_mass_simulated(alpha3, m_a, m_b, f_bin_simulated)
+                  / mean_mass_of_binary(f0, f_bin_nature, m_1, m_2, m_min, m_max,
+                                        alpha1, alpha2, alpha3))
+
+        underlying_total_mass=initial_ZAMS_TOTAL_binaries/f_corr_binaries
+
+        f_corr_single_stars=0
+
+    if (f_bin_simulated != 1): #when you have modeled both binary and single stars) 
+        if (f_bin_nature == 0): #you want the underlying mass for a population consisting of only single stars
+
+            
+
+            f_corr_single_stars = (fraction_simulated(1, m_1, m_2, m_min, m_max, alpha1, alpha2, alpha3)
+                  * mean_mass_simulated(alpha3, m_a, m_b, f_bin_simulated)
+                  / mean_mass_of_binary(f0, f_bin_nature, m_1, m_2, m_min, m_max,
+                                        alpha1, alpha2, alpha3))
+            
+            underlying_total_mass=initial_ZAMS_TOTAL_single/f_corr_single_stars
+            f_corr_binaries=0
+            
+        if (f_bin_nature == 1): #you want the underlying mass for a population consisting of only binary stars
+            f_corr_binaries = (fraction_simulated(1, m_1, m_2, m_min, m_max, alpha1, alpha2, alpha3)
+                  * mean_mass_simulated(alpha3, m_a, m_b, f_bin_simulated)
+                  / mean_mass_of_binary(f0, f_bin_nature, m_1, m_2, m_min, m_max,
+                                        alpha1, alpha2, alpha3))
+            
+            underlying_total_mass=initial_ZAMS_TOTAL_binaries/f_corr_binaries
+            f_corr_single_stars=0
+
+       
+        else:  #you want the underlying mass for a population consisting of both single and binary stars    
```

Comment:
Why do we need the other cases? Wouldn't it be OK, to just go with this general case always?